DependencyInjection/DoctrineMigrationsExtension.php

```diff
@@ -25,14 +34,92 @@ public function load(array $configs, ContainerBuilder $container) : void
+        }
+
+        foreach ($config['migrations'] as $migrationClass) {
+            $configurationDefinition->addMethodCall('addMigrationClass', [$migrationClass]);
+        }
+
+        if ($config['organize_migrations'] !== false) {
+            $configurationDefinition->addMethodCall('setMigrationOrganization', [$config['organize_migrations']]);
+        }
+
+        if ($config['custom_template'] !== null) {
+            $configurationDefinition->addMethodCall('setCustomTemplate', [$config['custom_template']]);
+        }
+
+        $configurationDefinition->addMethodCall('setAllOrNothing', [$config['all_or_nothing']]);
+        $configurationDefinition->addMethodCall('setCheckDatabasePlatform', [$config['check_database_platform']]);
+
+        $diDefinition = $container->getDefinition('doctrine.migrations.di');
+
+        if ($config['sorter'] !== null) {
+            $diDefinition->addMethodCall('setService', [DependencyFactory::MIGRATIONS_SORTER, new Reference($config['sorter'])]);
+        }
+
+        if ($config['storage']['id'] !== null) {
+            $diDefinition->addMethodCall('setService', [MetadataStorage::class, new Reference($config['storage']['id'])]);
+        } else {
+            $storageConfiguration = $config['storage']['table_storage'];
+
+            $storageDefinition = new Definition(TableMetadataStorageConfiguration::class);
+            $container->setDefinition('doctrine.migrations.storage.table_storage', $storageDefinition);
+            $container->setAlias('doctrine.migrations.metadata_storage', 'doctrine.migrations.storage.table_storage');
+
+            if ($storageConfiguration['table_name']!== null) {
+                $storageDefinition->addMethodCall('setTableName', [$storageConfiguration['table_name']]);
+            }
+            if ($storageConfiguration['version_column_name']!== null) {
+                $storageDefinition->addMethodCall('setVersionColumnName', [$storageConfiguration['version_column_name']]);
+            }
+            if ($storageConfiguration['version_column_length']!== null) {
+                $storageDefinition->addMethodCall('setVersionColumnLength', [$storageConfiguration['version_column_length']]);
+            }
+            if ($storageConfiguration['executed_at_column_name']!== null) {
+                $storageDefinition->addMethodCall('setExecutedAtColumnName', [$storageConfiguration['executed_at_column_name']]);
+            }
+            if ($storageConfiguration['execution_time_column_name']!== null) {
+                $storageDefinition->addMethodCall('setExecutionTimeColumnName', [$storageConfiguration['execution_time_column_name']]);
+            }
+
+            $configurationDefinition->addMethodCall('setMetadataStorageConfiguration', [new Reference('doctrine.migrations.storage.table_storage')]);
+        }
+
+        if ($config['em']!==null && $config['connection']!==null) {
```

Comment:
Spacing is wrong, but feel free to ignore this for now and enable doctrine/coding-standard separately.